lib/fileset/tests.sh

```diff
@@ -1439,6 +1439,16 @@ if [[ -n "$fetchGitSupportsSubmodules" ]]; then
 fi
 rm -rf -- *
 
+# shallow = true is not supported on all Nix versions
+if [[ "$(nix-instantiate --eval --expr "$prefixExpression (versionAtLeast builtins.nixVersion _fetchGitShallowMinver)")" == true ]]; then
+    createGitRepo full
+    git clone --depth 1 "file://${PWD}/full" shallow
+    cd shallow || die "Couldn't create a shallow clone of a Git repository."
```

Comment:
```suggestion
    # Extra commit such that there's a commit that won't be in the shallow clone
    git -C full commit --allow-empty -q -m extra
    git clone -q --depth 1 "file://${PWD}/full" shallow
    cd shallow
```

- Without the extra commit, older versions could also run this test successfully
- Add `-q` to make the `git clone` silent, such that the script doesn't output anything in the successful case
- No need for the `|| die`, because if `git clone` fails, it will return a non-0 exit code, which will stop the script because `set -e` is set

Reply:
- Actually not. I tested it and it fails even if there is only one commit. However, it still makes sense to create a second commit just to be sure and explicit.
- Yeah, makes sense. I didn't notice any additional output.
- Oh. Yeah, I read `set -e` and still forgot about it. I made some bad experiences with a failing `cd` and `rm` afterwards in the past. :)